packages/blockchain-wallet-v4-frontend/src/modals/Swap/EnterAmount/Checkout/index.tsx

```diff
@@ -182,49 +142,37 @@ const Checkout: React.FC<InjectedFormProps<{}, Props> & Props> = props => {
 
   const quoteAmount =
     fix === 'FIAT'
-      ? Exchange.convertFiatToCoin(
-          formValues?.amount || 0,
-          BASE.coin,
-          walletCurrency,
-          baseRates
-        )
-      : Exchange.convertCoinToFiat(
-          formValues?.amount || 0,
-          BASE.coin,
-          walletCurrency,
-          baseRates
-        )
+      ? Exchange.convertFiatToCoin(formValues?.amount || 0, BASE.coin, walletCurrency, baseRates)
+      : Exchange.convertCoinToFiat(formValues?.amount || 0, BASE.coin, walletCurrency, baseRates)
 
   const quoteAmountString =
     fix === 'FIAT'
       ? coinToString({
-          value: quoteAmount,
-          unit: { symbol: coins[BASE.coin].coinTicker }
+          unit: { symbol: coins[BASE.coin].coinTicker },
+          value: quoteAmount
         })
-      : fiatToString({ value: quoteAmount, unit: walletCurrency })
+      : fiatToString({ unit: walletCurrency, value: quoteAmount })
 
   const handleMinMaxClick = () => {
-    const value =
-      fix === 'FIAT'
-        ? amtError === 'BELOW_MIN'
-          ? fiatMin
-          : fiatMax
-        : amtError === 'BELOW_MIN'
-        ? min
-        : max
-    formActions.change('swapAmount', 'amount', value)
+    if (amtError === 'BELOW_MIN') {
+      props.swapActions.handleSwapMinAmountClick(fix === 'FIAT' ? fiatMin : min)
+    }
+
+    if (amtError === 'ABOVE_MAX' || amtError === 'NEGATIVE_INCOMING_AMT') {
```

Comment:
what if `amtError` is none of these three options, is not running `handleSwapMinAmountClick` ok at that point?

Reply:
The type only accepts these values